### DISCUSSION TITLE
Is there any example of how to store the component current changes in localstorage?

### DISCUSSION THREAD

**Discussion by RajathShanmugaiah** [react-page/react-page#1041] [Q&A]
I'm currently working on dynamic changes on certain components.

Im able to store the data in the local storage. 

I wanted to know how to load the the data on to the component from the localstorage .

Currently when I do this everytime the local storage refreshes.

**Comment by macrozone** [comment]
@RajathShanmugaiah Can you give more information and maybe some example code?

**Reply by RajathShanmugaiah** [reply]
@macrozone 
Here is my editor which is wrapped with the context
`<UserProvider value={{menu:apiMenuData, footer:footerData, website: website, setWebsite: setWebsite}}>
      <Editor languages={LANGUAGES} cellPlugins={cellPlugins} value={value} blurGateDisabled={true} onChange={setValue}/>
</UserProvider>`

website and setwebsite is state that stores the data and sets the localstorage

Here is my individual component that Im adding to **website**  state and setting the local storage 
`        useEffect(() => {
            if(website.length === 0){
                setWebsite([... website, {component:'hero', data:data}])
                localStorage.setItem('website', JSON.stringify(website))
            }
            else {
                let localWebIndex = website.findIndex((item: { component: string; }) => item.component === 'hero' )
                website[localWebIndex].data = data
                setWebsite( website.map((item: { component: string; }) => item.component === 'hero'? {component:'hero', data:data} : item))
            }
        },[website])`

so after doing this when I refresh the page the component is gone and when I again add the component the local storage refreshes and erases the old data so, On page load I still want the editor to load from the localStorage

`// In Renderer 
<TextBlockComponent textBlockData={data} />
`
so Basically if localStorage data is available I want to load something like this below in the component as well as the bottom toolbar
 `<TextBlockComponent textBlockData={website.length > 0? website | data} />`



I have another Issue now. Is there any way that I can have custom function written for the **remove plugin button in Bottom ToolBar** and **addblock in the control buttons** ?
Currently The **Remove Plugin** **onClick** removes the the component I want additional function to be written to it . Please let me know if there is a way out for it.
